V4

danhenderson opened the pull request.

rowanhogan commented on source/api/fitness-activities/justgiving/index.js:
```diff
@@ -1,51 +1,83 @@
-  message: activity.Title,
-  page: activity.PageGuid,
-  slug: activity.PageShortName,
-  source: activity.ActivityType
```
Both `source` && `sourceUrl` are used in SB, so would be good to add them back in below (when applicable)

rowanhogan commented on source/api/fitness-activities/justgiving/index.js:
```diff
@@ -1,51 +1,83 @@
+    page: activity.PageGuid,
+    slug: activity.PageShortName,
+    teamId: activity.TeamGuid,
+    message: activity.Title || activity.title,
```
It's a tiny pet peeve, but I find these big methods are so much easier to read when the keys are sorted alphabetically.

danhenderson replied:
I'm usually the same. I think I had "title" on my mind there.

rowanhogan commented on source/api/fitness-activities/justgiving/index.js:
```diff
@@ -56,9 +88,65 @@ export const fetchFitnessActivities = (params = required()) => {
+              createdAt
+              fitnessActivity {
+                title
+                activityType
```
`activityId` will be required per what I mentioned above, but it's also a part of the `legacyId` string, which includes Strava if applicable, e.g. `"FITNESS:STRAVA:3903099757"`

rowanhogan reviewed (changes requested):
Just a couple points

danhenderson commented:
As mentioned in Slack, there were a few things tripping me up, so I'm trying to consolidate things a little bit. Here are the main changes since you last saw it.

**Title and description**

We are in a weird state where title and description were mixed up between Strava and Manual, and they were also called different things (e.g. message, even though it was the title, not the message...very confusing).

I went through and made everything consistently title and description (I did allow caption to passed as a fallback for title just for a bit of backwards compatibility). I also updated the create fitness form to be able to `includeDescription` and `includeTitle`.

**Activity Type vs Type**

This was another that was mixed, this time between the Consumer API response and GraphQL response. So I moved them explicitly up to the top to set the `source` and `activityType` correctly.

**ID vs External ID vs Legacy ID**

Again, this was getting very confusing between Strava/Manual (via GraphQL or Consumer, depending on whether it is being returned by GraphQL or Consumer API. I have standardised this so regardless of if you are fetching from GraphQL or Consumer API (legacy), you get the same IDs in the same fields each time.

`id`: this is the timeline id (null in the case of legacy only manual fitness)
`externalId`: this is the Strava ID
`legacyId`: this is the Consumer API id (only available if `useLegacy: true` and can be used to delete old activities)

NOTE: most of the above is specific to the JG implementation, and the only change to EDH is to follow suit w/ title and description. The id is still always the fitness entry id for EDH (ahh, simpler times).

rowanhogan reviewed (approved):
👍
